.github/workflows/build.yaml

```diff
@@ -0,0 +1,48 @@
+name: Main Build
+
+on:
+    workflow_dispatch:
+    pull_request:
+        branches: [main]
+env:
+  GPG_PASSPHRASE: ${{secrets.GPG_PASSPHRASE}}
+  GPG_PRIVATE_KEY: ${{secrets.GPG_PRIVATE_KEY}}
+
+jobs:
+    build-push-wrapping-artefacts:
+        name: Building and Deploying Maven artefcats
+        runs-on: ubuntu-latest
+
+        steps:
+          - name: Checkin Code
+            uses: actions/checkout@v4
+          
+          - name: Set up JDK 11
+            uses: actions/setup-java@v4
+            with:
+              java-version: '11'
+              distribution: 'temurin'
+              server-id: 'github'
+              server-username: GITHUB_ACTOR
+              server-password: GITHUB_TOKEN
+              settings-path: '~/.m2'
+              gpg-private-key: ${{secrets.GPG_PRIVATE_KEY}}
+              gpg-passphrase: GPG_PASSPHRASE
+              cache: maven
+
+
+          - name: Building artefacts
+            env:
+              GPG_PASSPHRASE: ${{secrets.GPG_PASSPHRASE}}
+              GPG_PRIVATE_KEY: ${{secrets.GPG_PRIVATE_KEY}}
+              GITHUB_ACTOR: ${{github.actor}}
+              GITHUB_TOKEN: ${{github.token}}
+            run: |
+             mvn -X --batch-mode -Dgpg.passphrase=GPG_PASSPHRASE deploy \

```

Comment:
What does `-X` mean?

Reply:
It prints a more detailed trace of the build, we can remove this once errors are fixed.